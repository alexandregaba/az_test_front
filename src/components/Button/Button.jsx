import React from 'react';
import styled from 'styled-components';

const InnerButton = styled.button`
  height: 1.5em;
  min-width: 5em;
  margin: 3px 0;
  border: none;
  &:focus {
    outline: 0;
  }
  justify-self: space-between;
  background: transparent;
  &:hover {
    transform: translateY(-1px);
  }
  &:active {
    transform: translateY(1px);
  }
`;

const EditButton = styled(InnerButton)`
  margin-left: 10px;
`;

const DeleteButton = styled(InnerButton)`
  margin-right: 0;
`;

const Button = ({ type, editMode, handleEdit, handleDelete }) => {
  if (type === 'edit') {
    return (
      <EditButton className="editButton" onClick={handleEdit}>
        {editMode ? 'Save' : 'Edit'}
      </EditButton>
    );
  }
  return (
    <DeleteButton className="editButton" onClick={handleDelete}>
      Delete
    </DeleteButton>
  );
};

export default Button;
