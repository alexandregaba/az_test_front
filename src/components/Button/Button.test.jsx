import React from 'react';
import Button from './Button';
import { shallow } from 'enzyme';

const props = {
  type: 'edit',
  editMode: false,
  handleEdit: jest.fn(),
  handleDelete: jest.fn(),
};

it('matches snapshots', () => {
  const EditButton = shallow(<Button {...props} />);
  expect(EditButton).toMatchSnapshot();

  const DeleteButton = shallow(<Button {...props} type={'delete'} />);
  expect(DeleteButton).toMatchSnapshot();

  const EditButtonEditMode = shallow(<Button {...props} type={'edit'} editMode />);
  expect(EditButtonEditMode).toMatchSnapshot();
});

it('edit button triggers handleEdit', () => {
  const EditButton = shallow(<Button {...props} />);
  EditButton.simulate('click');
  expect(props.handleEdit).toHaveBeenCalled();
});

it('delete button triggers handleEdit', () => {
  const DeleteButton = shallow(<Button {...props} type={'delete'} />);
  DeleteButton.simulate('click');
  expect(props.handleDelete).toHaveBeenCalled();
});
