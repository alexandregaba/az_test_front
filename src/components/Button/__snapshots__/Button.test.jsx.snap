// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<Styled(styled.button)
  className="editButton"
  onClick={[MockFunction]}
>
  Edit
</Styled(styled.button)>
`;

exports[`matches snapshots 2`] = `
<Styled(styled.button)
  className="editButton"
  onClick={[MockFunction]}
>
  Delete
</Styled(styled.button)>
`;

exports[`matches snapshots 3`] = `
<Styled(styled.button)
  className="editButton"
  onClick={[MockFunction]}
>
  Save
</Styled(styled.button)>
`;
